src/lambdas/RedriveSNSDQLEvents/handler.ts

```diff
@@ -0,0 +1,50 @@
+import { Context, SQSBatchResponse, SQSEvent } from 'aws-lambda'
+import { initialiseLogger, logger } from '../../sharedServices/logger'
+import { AuditEvent } from '../../types/auditEvent'
+import {
+  generateEventIdLogMessageFromProcessingResult,
+  parseSQSEvent,
+  SQSBatchItemFailureFromProcessingResultArray
+} from './helper'
+import { writeToFirehose } from './writeToFirehose'
+
+export type ProcessingResult = {
+  sqsMessageId: string
+  failed: boolean
+  statusReason: string
+  auditEvent?: AuditEvent
+}
+
+export const handler = async (
+  event: SQSEvent,
+  context: Context
+): Promise<SQSBatchResponse> => {
+  initialiseLogger(context)
+
+  const { successfullyParsedRecords, unsuccessfullyParsedRecords } =
+    parseSQSEvent(event)
+
+  const firehoseResponse = await writeToFirehose(successfullyParsedRecords)
+
+  const unsuccessfullyParsedRecordsSQSMessageId =
+    SQSBatchItemFailureFromProcessingResultArray(unsuccessfullyParsedRecords)
+
+  const unsucessfullySentToFirehoseSQSMessageId =
+    SQSBatchItemFailureFromProcessingResultArray(
+      firehoseResponse.failedProcessingResults
+    )
+
+  const batchItemFailure = unsuccessfullyParsedRecordsSQSMessageId.concat(
+    unsucessfullySentToFirehoseSQSMessageId
+  )
+
+  logger.info('processed the following event ids', {
+    event_id: generateEventIdLogMessageFromProcessingResult([
+      firehoseResponse.failedProcessingResults,
```

Comment:
Is it possible for these to be spread into a single array? I think it would make the implementation of the function easier?

Reply:
my reasoning behind `generateEventIdLogMessageFromProcessingResult()` accepting an array is so that we can test that all potential failure cases can be included as part of the log message. alternative would be doing an `Array.concat(arr1, arr2)`. dont think spreading into a single array would make it any more easier imo